### PR TITLE
Update lilypond*

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change
- Rework both to use `fetchgit`
- ~~Use `rsync` in `lilypond` build~~
- Add me as maintainer of `lilypond` (and by extension `lilypond-unstable`)
- Mark `lilypond` broken on darwin (segfault on my machine, been failing on hydra awhile)
- Upgrade `lilypond-stable` from `2.19.65` to `2.19.83`
- Prefer `overrideAttrs` to `overrideDerivation` in `lilypond-unstable`
- Use fonts from `ghostscript` instead of pointing to a nonexistent urw-fonts repo
- Install fonts to the `lilypond` `datadir`
- Update LilyJAZZ font

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by 7c6f434c** [comment]
Hm, nice

**Comment by yurrriq** [comment]
@GrahamcOfBorg build lilypond
@GrahamcOfBorg build lilypond-unstable
@GrahamcOfBorg build lilypond-with-fonts

**Comment by 7c6f434c** [comment]
Any comments on how it uses `rsync`?

**Comment by yurrriq** [comment]
I added it blindly because the logs suggested it, but it looks like it's used when building documentation. http://lilypond.org/doc/v2.19/Documentation/contributor/requirements-for-building-documentation

We could make it optional with the default `null`, if we want.

**Comment by yurrriq** [comment]
@GrahamcOfBorg build lilypond

**Comment by yurrriq** [comment]
@GrahamcOfBorg build lilypond-unstable

**Comment by yurrriq** [comment]
@GrahamcOfBorg build lilypond-with-fonts

**Comment by yurrriq** [comment]
:tada: Thanks, @7c6f434c! Let me know if you have any on how to improve the `with-fonts` expression. I'd love to be able to replace usage like this:

```nix
self: super {
  lilypond-with-improviso = super.lilypond-with-fonts.override {
    fonts = with super.openlilylib-fonts; [ improviso ];
  };
}
```

with this:
```nix
self: super {
  lilypond-with-improviso = super.lilypond-with-fonts [ improviso ];
}
```

...but I don't know how to support it. It seems like `haskellPackages` and `idrisPackages` do something similar.

**Comment by 7c6f434c** [comment]
>```nix
>self: super {
>  lilypond-with-improviso = super.lilypond-with-fonts [ improviso ];
>}
>```

This is probably impossible as `improviso` has to be imported from its scope…

**Comment by yurrriq** [comment]
Ah, I see. It actually works like this:

```nix
self: super {
  my-idris = super.idrisPackages.with-packages (with super.idrisPackages; [ contrib pruviloj ]);
}
```

So maybe this would be a feasible goal:
```nix
self: super {
  lilypond-with-improviso = super.lilypond-with-fonts (with super.openlilylib-fonts; [ improviso ]);
}
```

**Comment by 7c6f434c** [comment]
>So maybe this would be a feasible goal:
>```nix
>self: super {
>  lilypond-with-improviso = super.lilypond-with-fonts (with super.openlilylib-fonts; [ improviso ]);
>}
>```

This is basically what 
```nix
lilypond-with-fonts = fonts: callPackage … { inherit fonts; };
```
does, I think?

File: pkgs/misc/lilypond/with-fonts.nix [hunk 1]
```diff
@@ -1,35 +1,18 @@
...
-  buildInputs = fonts;
-  nativeBuildInputs = [ lndir ];
-  installPhase = ''
-    local fontsdir=$out/share/lilypond/${lilypond.version}/fonts
+stdenv.lib.appendToName "with-fonts" (symlinkJoin {
+  inherit (lilypond) meta name version ;
 
-    install -m755 -d $fontsdir/otf
-    install -m755 -d $fontsdir/svg
+  paths = [ lilypond ];
```
Comment (yurrriq) [inline pkgs/misc/lilypond/with-fonts.nix]: I forgot `++ fonts` here...

**Pull request merged**